Breeze.UI/src/app/wallet/advanced/advanced.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { FormGroup, Validators, FormBuilder, AbstractControl } from '@angular/forms';
+import { NgbDateStruct } from '@ng-bootstrap/ng-bootstrap';
+
+import './monitor';
+import { AdvancedService } from './advanced.service';
+import { LoadingState } from './loadingState';
+import { SerialDisposable } from './serialDisposable';
+
+@Component({
+  selector: 'app-advanced',
+  templateUrl: './advanced.component.html',
+  styleUrls: ['./advanced.component.css']
+})
+export class AdvancedComponent implements OnInit, OnDestroy {
+    private addressCount = "";
+    private extPubKeySubs = new SerialDisposable();
+    private generateAddressesSubs = new SerialDisposable();
+    private resyncSubs = new SerialDisposable();
+    private addresses: string[] = [];
+    private resyncActioned = false;
+
+    constructor(private advancedService: AdvancedService, private formBuilder: FormBuilder) { 
+        this.setResyncDates();
+    }
+
+    public icoFormGroup: FormGroup;
+    public extPubKey = "";
+    public resyncDate: NgbDateStruct;
+    public extPubKeyLoadingState = new LoadingState();  
+    public generateAddressesLoadingState = new LoadingState();
+    public resyncLoadingState = new LoadingState();
+    public maxResyncDate: NgbDateStruct;
+    public minResyncDate: NgbDateStruct;
+    public get datePickerControl(): AbstractControl { return this.icoFormGroup.get('datePickerControl'); }
+    public get addressCountControl(): AbstractControl { return this.icoFormGroup.get('addressCountControl'); }
+    public get showAddressesTick(): boolean { return this.generateAddressesLoadingState.success && 
+                                            this.addresses.length && (Number(this.addressCount)===this.addresses.length); }
+    public get showResyncTick(): boolean { return this.resyncLoadingState.success && this.resyncActioned; }
+
+    ngOnInit() {
+        this.registerFormControls();
+        this.loadExtPubKey();
+    } 
+
+    resync() {
+        this.resyncActioned = true;
+        const date = new Date(this.resyncDate.year, this.resyncDate.month-1, this.resyncDate.day);
+        this.resyncSubs.disposable = this.advancedService.resyncFromDate(date)
+                                              .monitor(this.resyncLoadingState)
+                                              .subscribe();
+    }
+
+    generateAddresses() {
+        this.addresses = [];
```

Comment:
`this.addresses.length = 0` is a much faster way of clearing an array without memory re-assignment

Reply:
Sure, but performance isn't a concern here, and the reassignment reads better (personally).

Reply:
As far as I know it is a bad practice: https://davidwalsh.name/empty-array

Reply:
You might get unexpected behavior or memory leaks plus it future proofs it for any new usages of addresses array.

Reply:
@herbepau any update?